**Задача 4. Есть ли статистически значимые различия в росте дочерей?
Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165
Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160**


**Решение:**\
Выдвигаем гипотезы:\
**H<sub>0</sub>** - рост матерей и дочерей равен\
**H<sub>A</sub>** - рост матерей не равен росту дочерей

In [1]:
import numpy as np
import scipy.stats as stats
mother = [172, 177, 158, 170, 178,175, 164, 160, 169, 165]
daugther = [173, 175, 162, 174, 175, 168, 155, 170, 160]
len(mother) == len(daugther)

False

У приведенных в условии массивах неравное кол-во элементов, поэтому к ним нельзя применить функции ttest_rel, поэтому я решил немного изменить условия и добавил в меньший массив (выборка по дочерям) еще одно медианное значение.

In [2]:
m = np.mean(daugther)
daugther.append(m)
print(daugther)

[173, 175, 162, 174, 175, 168, 155, 170, 160, 168.0]


Теперь можем применять встроенную функцию:

In [3]:
test = stats.ttest_rel(mother,daugther)
print(test)

Ttest_relResult(statistic=0.40022240757904204, pvalue=0.6983260526026128)


Вывод: Поскольку значение p-value (0.6983) больше 0.05, полученные данные являются статистически значимыми, т.е. мы отвергаем **H<sub>0</sub>** в пользу **H<sub>A</sub>** и делаем вывод, росто взрослых дочерей никак не зависит от роста их матерей